packages/playwright-core/src/protocol/serializers.ts

```diff
@@ -184,5 +184,6 @@ function isURL(obj: any): obj is URL {
 }
 
 function isError(obj: any): obj is Error {
-  return obj instanceof Error || obj?.__proto__?.name === 'Error' || (obj?.__proto__ && isError(obj.__proto__));
+  const proto = obj ? Object.getPrototypeOf(obj) : null;
```

Comment:
This will fail if there is a user code like `Object.getPrototypeOf = () => {};`

Reply:
Oh, this is a server side code, we don't run random js here so it should be ok.

Reply:
Indeed, and `obj.__proto__` was an order of magnitude less safe than Object.getPrototypeOf :)